### PR TITLE
[api][backend] freeze package sources on release

### PR DESCRIPTION
**Pull request by adrianschroeter**
feel free to review, but please do not merge yet, further checking is needed.

### CODE REVIEW

File: src/api/app/helpers/branch_package.rb [hunk 1]
```diff
@@ -149,8 +149,11 @@ def create_branch_packages(tprj)
         opackage = p[:package].name if p[:package].is_a? Package
         oproject = p[:link_target_project]
         oproject = p[:link_target_project].name if p[:link_target_project].is_a? Project
+        linkrev=nil
```
Comment (bgeuken) [inline src/api/app/helpers/branch_package.rb]: linkrev = nil please

File: src/api/app/helpers/maintenance_helper.rb [hunk 2]
```diff
@@ -150,11 +150,12 @@ def release_package_copy_sources(action, sourcePackage, targetPackageName, targe
...
     }
     cp_params[:requestid] = action.bs_request.id if action
+    cp_params[:freezelink] = 1 if sourcePackage.is_link? # ne permission check here on purpose
     cp_path = "/source/#{CGI.escape(targetProject.name)}/#{CGI.escape(targetPackageName)}"
     cp_path << Suse::Backend.build_query_from_hash(cp_params, [:cmd, :user, :oproject,
                                                                :opackage, :comment, :requestid,
                                                                :expand, :withvrev, :noservice,
-                                                               :withacceptinfo])
+                                                               :freezelink, :withacceptinfo])
     result = Suse::Backend.post cp_path, nil
```
Comment (bgeuken) [inline src/api/app/helpers/maintenance_helper.rb]: parenthesis please

File: src/api/app/models/package.rb [hunk 3]
```diff
@@ -968,7 +968,7 @@ def valid_name
     errors.add(:name, 'is illegal') unless Package.valid_name?(self.name)
   end
 
-  def branch_from(origin_project, origin_package, rev = nil, missingok = nil, comment = nil)
+  def branch_from(origin_project, origin_package, rev = nil, missingok = nil, comment = nil, linkrev = nil)
```
Comment (bgeuken) [inline src/api/app/models/package.rb]: That might be a good point to replace some of the parameters by an options hash:)

File: src/api/app/helpers/branch_package.rb [hunk 4]
```diff
@@ -149,8 +149,11 @@ def create_branch_packages(tprj)
         opackage = p[:package].name if p[:package].is_a? Package
         oproject = p[:link_target_project]
         oproject = p[:link_target_project].name if p[:link_target_project].is_a? Project
+        linkrev = nil
+        linkrev = params[:linkrev] if params[:linkrev]
```
Comment (hennevogel) [inline src/api/app/helpers/branch_package.rb]: If `params[:linkrev]` isn't it's nil...

File: src/api/app/helpers/maintenance_helper.rb [hunk 5]
```diff
@@ -150,11 +150,12 @@ def release_package_copy_sources(action, sourcePackage, targetPackageName, targe
       withacceptinfo: "1"
     }
     cp_params[:requestid] = action.bs_request.id if action
+    cp_params[:freezelink] = 1 if sourcePackage.is_link? # ne permission check here on purpose
```
Comment (hennevogel) [inline src/api/app/helpers/maintenance_helper.rb]: s/ne/no/

File: src/api/app/models/package.rb (hunk 3 above)
Comment (adrianschroeter) [inline src/api/app/models/package.rb]: indeed.

**Pull request merged**